libbeat/common/transport/tlscommon/tls_config.go

```diff
@@ -186,14 +230,18 @@ func makeVerifyConnection(cfg *TLSConfig) func(tls.ConnectionState) error {
 	case VerifyStrict:
 		if len(cfg.CASha256) > 0 {
 			return func(cs tls.ConnectionState) error {
+				if cfg.ESCAFingerprint != "" {
+					if err := trustESRootCA(cfg, cs.PeerCertificates); err != nil {
+						return err
+					}
+				}
```

Comment:
We probably should not do this here. If the user configured `ssl.verification_mode: strict` this means they already have all necessary certificates in place and we should should not touch their configuration.

Reply:
`VerifyFull` and `VerifyStrict` is essentially the same. The only difference is that `VerifyFull` accepts hostnames in the CommonName section of certs. I think you did the right thing in the first place, adding self-signed certificates is correct.

Reply:
Cool